### DISCUSSION TITLE
Is the Vercel CLI promote endpoint correct, different endpoint that is used in the UI?

### DISCUSSION THREAD

**Discussion by mikedidomizio** [vercel/vercel#10811] [Help] [answered]
### Question

I'm trying to use Vercel CLI to promote a build to production. I am receiving a 500 error. I've found that the error is from this [line of code](https://github.com/vercel/vercel/blob/main/packages/cli/src/commands/promote/request-promote.ts#L46)
For those that don't click through, the URL in the code is `/v9/projects/${project.id}/promote/${deployment.id}`. I logged the parameters I'm using within the code and found that the project ID I'm using is correct, and the deployment ID used is correct.

When I promote to production through the Vercel UI, it uses the `/v13/deployments` endpoint with `target: 'production'`. I manually updated the CLI locally and it works as expected.

When I look through the [Vercel REST API endpoints documentation](https://vercel.com/docs/rest-api/endpoints), there's no reference of `/promote/` in any place there, nor was a reference to that endpoint when that code was introduced in May 2023.

Does the `/v9/projects/${project.id}/promote/${deployment.id}` endpoint exist and is it the correct endpoint for promoting builds? I would expect if it was correct but using it wrong it wouldn't be a `5XX` error.

Searching for [similar](https://github.com/orgs/vercel/discussions/3474) issues makes me think it should possibly be /v13/deployments

Example
No response

Steps to Reproduce
Using vercel CLI, take an existing build and run it through the promote command.

```
vercel promote {DEPLOYMENT_URL} --token=$VERCEL_TOKEN --yes
```

Expected Results:

Build is promoted to production

Actual Results:

500 error

**Comment by cb1kenobi** [comment, marked as answer]
@mikedidomizio Hi! Thank you very much for bringing this issue to our attention. We have identified the cause of the 500 and the fix is being reviewed and will be deployed shortly.

I suspect once the fix is deployed, you may see a message that says "Could not find current production id for project". You may want to review your project's domain settings and possibly redeploy the project to make sure there's a "current" production deployment.

You are correct in that the `promote` endpoint is not documented and we'll have to get that added to the docs.

FYI, the difference between the `/v13/deployments` endpoint and the `promote` endpoint is the `promote` endpoint only assigns aliases. It doesn't rebuild the deployment. You need to be careful if you promote a "preview" deployment as it will essentially make the deployment a production deployment with preview build settings/environment variables.

Best of luck!
